Content.Server/Chat/ChatConditions/ActiveTelecomsEntityChatCondition.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Linq;
+using Content.Server.Power.Components;
+using Content.Server.Radio.Components;
+using Content.Server.Station.Components;
+using Content.Server.Station.Systems;
+using Content.Shared.Chat;
+using Content.Shared.Chat.Prototypes;
+using Content.Shared.Radio;
+using Content.Shared.Radio.Components;
+using Robust.Shared.Network;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Chat.ChatConditions;
+
+/// <summary>
+/// Checks if the consumer is on a map with an active server, if necessary.
+/// </summary>
+[DataDefinition]
+public sealed partial class ActiveTelecomsEntityChatCondition : EntityChatCondition
+{
+
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IEntityManager _entityManager = default!;
+
+    public override HashSet<EntityUid> FilterConsumers(HashSet<EntityUid> consumers, Dictionary<Enum, object> channelParameters)
+    {
+        IoCManager.InjectDependencies(this);
```

Comment:
a thing to consider - maybe its possible to do DI inject on c-tor? Are those re-created often? It feels like it should most likely stay alive whole game session!